### DISCUSSION TITLE
Progress Task Description fixed column width

### DISCUSSION THREAD

**Discussion by stuartstein777** [spectreconsole/spectre.console#241] [Q&A] [answered]
Hi,

I'm updating a task that shows the description field. I want to display the item being processed but they all have a different length. I've tried padding the string in `task.Description()` to a minimum that I think it will fit in, but it seems like Spectre trims it?

The result is as each time the Description changes the progress bar and percentage move left and right.

Is their something I'm missing that would fix the width of the TaskDescription column ?

A little reproducible example

```
static async Task Main(string[] args)
		{
			var items = new List<string>
						{
							"abc", "def", "ddsfasdgsdg", "dsgjksdgjgospdgopj", "ogodsjg", "odsgj", "sjgosdpjgopsdjgo", "fgdg", "fgg4t43gt43gsg",
							"abc", "def", "ddsfasdgsdg", "dsgjksdgjgospdgopj", "ogodsjg", "odsgj", "sjgosdpjgopsdjgo", "fgdg", "fgg4t43gt43gsg",
							"abc", "def", "ddsfasdgsdg", "dsgjksdgjgospdgopj", "ogodsjg", "odsgj", "sjgosdpjgopsdjgo", "fgdg", "fgg4t43gt43gsg",
							"abc", "def", "ddsfasdgsdg", "dsgjksdgjgospdgopj", "ogodsjg", "odsgj", "sjgosdpjgopsdjgo", "fgdg", "fgg4t43gt43gsg",
							"abc", "def", "ddsfasdgsdg", "dsgjksdgjgospdgopj", "ogodsjg", "odsgj", "sjgosdpjgopsdjgo", "fgdg", "fgg4t43gt43gsg"
						};
			await AnsiConsole.Progress()
							 .Columns(new ProgressColumn[]
								{
									new TaskDescriptionColumn(),
									new ProgressBarColumn(),
									new PercentageColumn()
								})
							 .StartAsync(async ctx =>
										{
											var task1 = ctx.AddTask("[blue]foo[/b]");
											task1.MaxValue = items.Count;
											foreach (var item in items)
											{
												task1.Description = item;
												task1.Increment(1.0);
												await Task.Delay(150);
											}
										});
		}
```

Thanks!

**Comment by patriksvensson** [comment, marked as answer]
@stuartstein777 There is no way of currently doing this, but if this is something you need, please open an issue.
In the meanwhile, you might want to move the text column to the far right instead to avoid the "flickering".

![progresscolumns](https://user-images.githubusercontent.com/357872/105205449-92e0a900-5b45-11eb-9a9d-ec11ea924259.gif)
